backend/apis/leads/add-leads.api.js

```diff
@@ -96,9 +98,12 @@ app.post("/api/leads", (req, res) => {
           leadDataArray.push(eventId);
         }
       }
+
+      leadEventsQuery += ";";
     }
 
-    const newLeadQuery = `${addToLeadQuery}; ${setLeadId}; ${leadServicesQuery}; ${leadEventsQuery};`;
```

Comment:
This query didn't work because it would put semicolons in for all the queries, even if the queries were empty. Mysql doesn't like it when it sees two semicolons right next to each other.

Specifically this would throw an error when the lead services array was passed in as empty.